src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export function isValidSides(a: number, b = 1, c = 1): boolean {
+  if (a <= 0 || b <= 0 || c <= 0) {
+    return true;
+  }
 
+  return false;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    readonly shape: Shape = 'triangle',
+  ) {
+    const sideA = this.a;
+    const sideB = this.b;
+    const sideC = this.c;
+
+    const isTriangle = sideA >= sideB + sideC
+      || sideB >= sideA + sideC
+      || sideC >= sideA + sideB;
+
+    if (isValidSides(sideA, sideB, sideC)) {
```

Comment:
Maybe `isInvalidSides`. The condition looks strange: if the parties are valid, issue an error